test/test.hapi-pagedata.js

```diff
@@ -0,0 +1,296 @@
+    server = new Hapi.Server({
+      debug: { log: ['error', 'info', 'hapi-pagedata'], request: ['error'] }
+    });
+    server.connection({ port: 8080 });
+    done();
+  });
+  lab.afterEach((done) => {
+    server.stop(done);
+  });
+  lab.test('set', allDone => {
+    async.auto({
+      register: (done) => {
+        server.register({
+          register: require('../'),
+          options: {
+            host: 'http://localhost:8080',
+            key: 'key',
+            cacheEndpoint: '/cache',
+            hookEndpoint: '/hook',
+            verbose: true,
+            site: 'site',
+            tag: 'test'
+          }
+        }, done);
+      },
+      api: ['register', (done, results) => {
+        expect(server.plugins['hapi-pagedata'].api).to.exist;
+        expect(server.plugins['hapi-pagedata'].api.PageData).to.exist;
+        expect(typeof server.plugins['hapi-pagedata'].api.options).to.equal('object');
+        expect(server.plugins['hapi-pagedata'].api.options.key).to.equal('key');
+        done();
+      }],
+      cache: ['api', (done, results) => {
+        expect(server.plugins['hapi-pagedata'].cache).to.exist;
+        expect(typeof server.plugins['hapi-pagedata'].cache.rule).to.equal('object');
+        expect(server.plugins['hapi-pagedata'].cache.rule.expiresIn).to.equal(1);
+        done();
+      }]
+    }, allDone);
+  });
+});
+
+lab.experiment('exposed server objects', () => {
+  lab.beforeEach(done => {
+    server = new Hapi.Server({
+      debug: { log: ['error', 'info', 'hapi-pagedata'], request: ['error'] }
+    });
+    server.connection({ port: 8080 });
+    done();
+  });
+  lab.afterEach((done) => {
+    server.stop(done);
+  });
+  lab.test('set', allDone => {
+    async.auto({
+      register: (done) => {
+        server.register({
+          register: require('../'),
+          options: {
+            host: 'http://localhost:8080',
+            key: 'key',
+            cacheEndpoint: '/cache',
+            hookEndpoint: '/hook',
+            verbose: true,
+            site: 'site',
+            tag: 'test'
+          }
+        }, done);
+      },
+      route: ['register', (done, results) => {
+        server.method('test', (cb) => {
+          setTimeout(() => {
+            cb(null, { some: 'thing' });
+          }, 500);
+        });
+        server.route({
+          path: '/',
+          method: 'GET',
+          config: {
+            pre: [
+              { method: 'test()', assign: 'test' }
+            ],
+          },
+          handler(request, reply) {
+            expect(typeof request.pre.test).to.equal('object');
+            expect(request.pre.test.some).to.equal('thing');
+            reply(request.pre);
+          }
+        });
+        server.start(done);
+      }],
+      verify: ['route', (done, results) => {
+        server.inject({
+          url: '/',
+          method: 'GET',
+        }, () => {
+          done();
+        });
+      }]
+    }, allDone);
+  });
+});
```

Comment:
can we add tests for the cache hooks?  I think there might be a bug somewhere in there or needs to be updated with the site params